### ISSUE TITLE
Operator gets killed because it runs out of memory

### ISSUE THREAD

**Issue by florianluediger** [hivemq/helm-charts#66]
When following [these instructions](https://www.hivemq.com/docs/operator/4.8/kubernetes-operator/operator-intro.html), I have noticed that the default settings for the operator do not provide enough memory to run it.

After running `helm upgrade --install hivemq hivemq/hivemq-operator`, the operator pod will be created but when monitoring with K9s, you can see that it crashes pretty soon due to an OOM error.

The pod will restart pretty quickly, however it does not seem to function properly after that.

I suggest increasing the default value for the operator memory limit.

**Comment by hurtadosanti** [comment]
Hi @florianluediger I will review this, we have not seen this behavior yet,  can you provide more context for the error, please?

**Comment by florianluediger** [comment]
Sure, just let me know what you need to know. Here is some more general information:

I have started minikube with 8192M of memory and 4 CPUs. 
Then I have executed `helm upgrade --install hivemq hivemq/hivemq-operator` with no special values provided.

Here is the log of the first start of the hivemq operator pod (grabbed from k9s):
```
Picked up JAVA_TOOL_OPTIONS: -XX:+UnlockExperimentalVMOptions -XX:InitialRAMPercentage=30 -XX:MaxRAMPercentage=80 -XX:MinRAMPercentage=30                                                                       │
09:40:00.261 [main] INFO  com.hivemq.Application - Preparing SSL files                                                                                                                                          │
09:40:08.758 [main] INFO  com.hivemq.Application - Wrote converted key store to /tmp/store.p12                                                                                                                  │
 __  __ _                                  _                                                                                                                                                                    │
|  \/  (_) ___ _ __ ___  _ __   __ _ _   _| |_                                                                                                                                                                  │
| |\/| | |/ __| '__/ _ \| '_ \ / _` | | | | __|                                                                                                                                                                 │
| |  | | | (__| | | (_) | | | | (_| | |_| | |_                                                                                                                                                                  │
|_|  |_|_|\___|_|  \___/|_| |_|\__,_|\__,_|\__|                                                                                                                                                                 │
  Micronaut (v2.4.2)                                                                                                                                                                                            │
                                                                                                                                                                                                                │
Stream closed EOF for default/hivemq-hivemq-operator-operator-6fdb648d47-pbcbb (operator)
```

The log of the operator pod after restarting the pod is empty. It seems like after the restart, the micronaut service does not start anymore.

Here is the output of `kubectl describe pod hivemq-hivemq-operator-operator-6fdb648d47-xv6rg`:

```
Name:         hivemq-hivemq-operator-operator-6fdb648d47-xv6rg
Namespace:    default
Priority:     0
Node:         minikube/192.168.49.2
Start Time:   Wed, 10 Aug 2022 11:41:45 +0200
Labels:       app=hivemq-operator
              operator=hivemq-hivemq-operator
              pod-template-hash=6fdb648d47
Annotations:  <none>
Status:       Running
IP:           172.17.0.4
IPs:
  IP:           172.17.0.4
Controlled By:  ReplicaSet/hivemq-hivemq-operator-operator-6fdb648d47
Containers:
  operator:
    Container ID:   docker://3de23c5599f24957108136bceb2f5e4682a6c47ded9b16a4bcb83a4a6cd38dfd
    Image:          hivemq/hivemq-operator:4.7.1
    Image ID:       docker-pullable://hivemq/hivemq-operator@sha256:96faf436164ca20248587b37593308b9422bca71385063b3f2a4859b9a8666ef
    Port:           8443/TCP
    Host Port:      0/TCP
    State:          Running
      Started:      Wed, 10 Aug 2022 11:42:18 +0200
    Last State:     Terminated
      Reason:       OOMKilled
      Exit Code:    137
      Started:      Wed, 10 Aug 2022 11:41:45 +0200
      Finished:     Wed, 10 Aug 2022 11:42:17 +0200
    Ready:          True
    Restart Count:  1
    Limits:
      cpu:     800m
      memory:  512M
    Requests:
      cpu:     200m
      memory:  512M
    Environment:
      JAVA_TOOL_OPTIONS:       -XX:+UnlockExperimentalVMOptions -XX:InitialRAMPercentage=30 -XX:MaxRAMPercentage=80 -XX:MinRAMPercentage=30
      OPERATOR_LOG_LEVEL:      INFO
      CREATE_CRD:              false
      MICRONAUT_SERVER_PORT:   8443
      MICRONAUT_SSL_ENABLED:   true
      OPERATOR_SSL_PATH:       /etc/ssl/admission
      MICRONAUT_CONFIG_FILES:  /tmp/application.yml
      SSL_FQDN:                hivemq-operator-operator.default.svc
    Mounts:
      /etc/ssl/admission from ssl-secrets (rw)
      /templates from templates (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-tvw8l (ro)
Conditions:
  Type              Status
  Initialized       True 
  Ready             True 
  ContainersReady   True 
  PodScheduled      True 
Volumes:
  templates:
    Type:      ConfigMap (a volume populated by a ConfigMap)
    Name:      hivemq-hivemq-operator-operator-templates
    Optional:  false
  ssl-secrets:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  hivemq-hivemq-operator-admission
    Optional:    false
  kube-api-access-tvw8l:
    Type:                    Projected (a volume that contains injected data from multiple sources)
    TokenExpirationSeconds:  3607
    ConfigMapName:           kube-root-ca.crt
    ConfigMapOptional:       <nil>
    DownwardAPI:             true
QoS Class:                   Burstable
Node-Selectors:              <none>
Tolerations:                 node.kubernetes.io/not-ready:NoExecute op=Exists for 300s
                             node.kubernetes.io/unreachable:NoExecute op=Exists for 300s
Events:
  Type    Reason     Age                From               Message
  ----    ------     ----               ----               -------
  Normal  Scheduled  11m                default-scheduler  Successfully assigned default/hivemq-hivemq-operator-operator-6fdb648d47-xv6rg to minikube
  Normal  Pulled     10m (x2 over 11m)  kubelet            Container image "hivemq/hivemq-operator:4.7.1" already present on machine
  Normal  Created    10m (x2 over 11m)  kubelet            Created container operator
  Normal  Started    10m (x2 over 11m)  kubelet            Started container operator

```

**Comment by hurtadosanti** [comment]
Hello, @florianluediger. Thank you very much for your contribution. Unfortunately, your change is for a specific use case, so we will reject it for the time being. We suggest overriding the values using the command from the [official documentation](https://www.hivemq.com/docs/operator/latest/kubernetes-operator/deploying.html#helm-chart)  
  
`helm upgrade --install -f myCustomValues.yaml hivemq hivemq/hivemq-operator`  
  
We will also recommend using smaller cluster and node requirements or providing more memory to the VM 
  
``` 
hivemq:  
  nodeCount: "1" 
  cpu: "2" 
  memory: "2Gi" 
``` 
  
Best wishes,

**Comment by denniseffing** [comment]
@hurtadosanti I ran into the same issue and tbh I don't understand how this qualifies as a "specific use case". I tried installing the operator onto multiple clusters with its default configuration and k8s always ends up killing it due to an OOM issue.

Why are the default memory limits set to 512 MiB if 512 MiB are not sufficient for running the operator? I understand that I am able to override these limits but people who try out the operator expect it to run in its most basic default configuration first. This is obviously not the case.
